jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java

```diff
@@ -18,69 +18,135 @@
+    private final Pool<Connection> pool;
 
     /**
-     * @param destination the correspondent destination
-     * @param maxConnections the max number of connections
-     * @param requester the callback to notify about new connection creation/failure
-     * @deprecated use {@link #AbstractConnectionPool(HttpDestination, int, Callback)} instead
+     * @deprecated use {@link #AbstractConnectionPool(HttpDestination, int, boolean, Callback)} instead
      */
     @Deprecated
     protected AbstractConnectionPool(Destination destination, int maxConnections, Callback requester)
     {
-        this((HttpDestination)destination, maxConnections, requester);
+        this((HttpDestination)destination, maxConnections, true, requester);
     }
 
-    protected AbstractConnectionPool(HttpDestination destination, int maxConnections, Callback requester)
+    protected AbstractConnectionPool(HttpDestination destination, int maxConnections, boolean cache, Callback requester)
     {
         this.destination = destination;
-        this.maxConnections = maxConnections;
         this.requester = requester;
+        @SuppressWarnings("unchecked")
+        Pool<Connection> pool = destination.getBean(Pool.class);
+        if (pool == null)
+        {
+            pool = new Pool<>(maxConnections, cache ? 1 : 0);
+            destination.addBean(pool);
+        }
+        this.pool = pool;
+    }
+
+    @Override
+    public void preCreateConnections(int connectionCount)
+    {
+        List<Future<Connection>> futures = new ArrayList<>();
+        for (int i = 0; i < connectionCount; i++)
+        {
+            futures.add(tryCreate(pool.getMaxEntries()));
+        }
+        List<Throwable> exceptions = new ArrayList<>();
+        for (Future<Connection> future : futures)
+        {
+            try
+            {
+                future.get();
+            }
+            catch (Throwable t)
+            {
+                exceptions.add(t);
+            }
+        }
+        if (!exceptions.isEmpty())
+        {
+            RuntimeException runtimeException = new RuntimeException("Error pre-creating connections");
+            exceptions.forEach(runtimeException::addSuppressed);
+            throw runtimeException;
+        }
+    }
+
+    protected int getMaxMultiplex()
+    {
+        return pool.getMaxMultiplex();
+    }
+
+    protected void setMaxMultiplex(int maxMultiplex)
+    {
+        pool.setMaxMultiplex(maxMultiplex);
     }
 
-    protected HttpDestination getHttpDestination()
+    protected int getMaxUsageCount()
     {
-        return destination;
+        return pool.getMaxUsageCount();
+    }
+
+    protected void setMaxUsageCount(int maxUsageCount)
+    {
+        pool.setMaxUsageCount(maxUsageCount);
+    }
+
+    @ManagedAttribute(value = "The number of active connections", readonly = true)
+    public int getActiveConnectionCount()
+    {
+        return (int)pool.values().stream().filter(pc -> pc.getState() > 0).count();
```

Comment:
`getState()` is IMHO abstract enough that it can its contract can accomodate with a different implementation. But I do agree that having three explicit `boolean isClosed()`  `boolean isIdle()` and `boolean isInUse()` methods is much clearer.
Let's do this change.